tfs/src/main/java/hudson/plugins/tfs/listeners/JenkinsRunListener.java

```diff
@@ -29,6 +44,9 @@ public void onDeleted(final Run run) {
 
     @Override
     public void onStarted(final Run run, final TaskListener listener) {
+        Job currJob = run.getParent();
+        final String targetUrl = currJob.getAbsoluteUrl() + (currJob.getNextBuildNumber() - 1);
+        trySetPullRequestStatus(run, listener, GitStatusState.Pending, "Jenkins CI build started", targetUrl);
```

Comment:
Seems like the trySetPullRequestStatus should be on the JenkinsEventNotifier class since it calls out to TFS/VSTS. 
Perhaps the name could be something more like sendPullRequestStatusEvent.
I would get the information from the run object in this class and pass it as parameters to the method on the JenkinsEventNotifier, but if you need to pass the whole run object, it's not that bad.

Reply:
Will this set the same status for each completed job, regardless of whether they are the same?

Like the github pull request builder, I suggest adding a context string to the trigger.  The context string defaults to the job name (full path), but could optionally be set to something like "x86 Release Build and Test"

Reply:
Good suggestion. Moved it to JenkinsEventNotifier and pass only needed run object to it, instead of the whole run object.

public static void sendPullRequestBuildStatusEvent(final CommitParameterAction commitParameter, final GitStatusState buildState, final String buildDescription, final String targetUrl) {